### DISCUSSION TITLE
Fails to connect to Waveshare USB

### DISCUSSION THREAD

**Discussion by bmgmac** [nathanmarlor/foxess_modbus#655] [Support]
Hi, sorry for the stupid newbie question but can’t seem to find the answer as I look through previous posts 
I have a Fox FH3 invertor that I am trying to connect via a Waveshare USB having used twisted pairs in a Cat 6 cable to my HomeAss computer - laptop running baremetal HAOS. 
I keep getting this error message when trying to install the integration despite having checked the connections at each point and the continuity of the cable itself. It is the same cable that carries the comms from the grid CTclamp for the inverter on the other 2 twisted pairs.  Any help / guidance would be really welcome. 
![IMG_8507](https://github.com/user-attachments/assets/d8cb5c32-d26d-4559-9d1d-83b05dbe03f2)
![IMG_8509](https://github.com/user-attachments/assets/82ebccbe-338a-485e-9505-0a1c36f580b1)
![IMG_8510](https://github.com/user-attachments/assets/d7563320-d9eb-4e71-aa8c-2acffd058525)
![IMG_8507](https://github.com/user-attachments/assets/4ec22df2-a68a-41c3-b048-07212c1e6861)
![IMG_8509](https://github.com/user-attachments/assets/087cbf2d-1c66-421d-a660-3feab57edefc)
![IMG_8510](https://github.com/user-attachments/assets/7b1a2fc7-e039-4041-85ad-670dc4595603)

**Comment by canton7** [comment]
This means that we're sending bytes to the adapter, but we're not getting anything back. This is usually either a wiring problem, or the H3 configuration problem. 

Make sure that your Slave ID is set to 247 in the front panel menu of the H3.

**Reply by bmgmac** [reply]
Thank you for answering. 
Think I have checked this this, but will do so again as often told by my wife I only 'man look'!. I have connected to pins 1 and 2 on the inverter connector as per the wiki. That is correct isn't it?

**Reply by FozzieUK** [reply]
Yep those are the correct pins for an H3 inverter, as well as the slave ID (which is often set as 0 on an H3) also double check your wiring is straight through i.e. pin 1 (RS485A) goes to 'A' and pin 2 (RS485B) to 'B' on the Waveshare.
